### ISSUE TITLE
[BUG] bcc mode possible errorous code generation

### ISSUE THREAD

**Issue by ninkin** [dashingsoft/pyarmor#1032]
Hello,

I was wondering if you could assist me with generating bcc enabled code using pyarmor pro, as I seem to be encountering an error. Would it be possible for you to take a look and offer some guidance?

Thank you very much for your help.

```
vscode ➜ /workspaces $ poetry run pyarmor gen --exclude tools --enable bcc --recursive  main.py
INFO     Python 3.8.10
INFO     Pyarmor 8.0.19 (pro)
INFO     search inputs ...
INFO     find script main.py
INFO     find  top resources
INFO     start to generate runtime files
INFO     target platforms {'linux.x86_64'}
INFO     patch runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "main"
INFO     obfuscating file main.py
INFO     start to generate bcc code...
.pyarmor/bcc/main.c: In function ‘_dictcomp_70_24’:
.pyarmor/bcc/main.c:8:33: error: ‘f_lineno’ undeclared (first use in this function)
    8 | # define    TRACE_LINE(x)       f_lineno = x
      |                                 ^~~~~~~~
.pyarmor/bcc/main.c:374:69: note: in expansion of macro ‘TRACE_LINE’
  374 |     if (!(x = ({PyObject *temp = ld_value(f_vars[17]); if (!temp) { TRACE_LINE(70); break; } temp; }))) {
      |                                                                     ^~~~~~~~~~
.pyarmor/bcc/main.c:8:33: note: each undeclared identifier is reported only once for each function it appears in
    8 | # define    TRACE_LINE(x)       f_lineno = x
      |                                 ^~~~~~~~
.pyarmor/bcc/main.c:374:69: note: in expansion of macro ‘TRACE_LINE’
  374 |     if (!(x = ({PyObject *temp = ld_value(f_vars[17]); if (!temp) { TRACE_LINE(70); break; } temp; }))) {
      |                                                                     ^~~~~~~~~~
.pyarmor/bcc/main.c:374:85: error: break statement not within loop or switch
  374 |     if (!(x = ({PyObject *temp = ld_value(f_vars[17]); if (!temp) { TRACE_LINE(70); break; } temp; }))) {
      |                                                                                     ^~~~~
.pyarmor/bcc/main.c:396:161: error: ‘m_consts’ undeclared (first use in this function); did you mean ‘ctx_consts’?
  396 |         if (wr_item(ld_value(r), ({PyObject *temp = ld_attr(({PyObject *temp = ld_value(f_targets[2]); if (!temp) { TRACE_LINE(70); break; } temp; }), ld_value(m_consts[43])); if (!temp) { TRACE_LINE(70); break; } temp; }), ({PyObject *temp = op_call(NULL, ({PyObject *temp = ld_attr(({PyObject *temp = op_call(NULL, ({PyObject *temp = ld_attr(({PyObject *temp = op_call(NULL, ({PyObject *temp = ld_attr(({PyObject *temp = ld_cell(f_vars[41]); if (!temp) { TRACE_LINE(70); break; } temp; }), ld_value(m_consts[49])); if (!temp) { TRACE_LINE(70); break; } temp; }), 1, 0, op_build(1, 1, ({PyObject *temp = ld_global(f_vars, ld_value(m_consts[52])); if (!temp) { TRACE_LINE(70); break; } temp; })), NULL); if (!temp) { TRACE_LINE(70); break; } temp; }), ld_value(m_consts[53])); if (!temp) { TRACE_LINE(70); break; } temp; }), 1, 0, op_build(1, 1, ({PyObject *temp = (x=op_compare(f_vars, PyCmp_EQ, ({PyObject *temp = ld_attr(({PyObject *temp = ld_global(f_vars, ld_value(m_consts[52])); if (!temp) { TRACE_LINE(70); break; } temp; }), ld_value(m_consts[43])); if (!temp) { TRACE_LINE(70); break; } temp; }), ({PyObject *temp = ld_attr(({PyObject *temp = ld_value(f_targets[2]); if (!temp) { TRACE_LINE(70); break; } temp; }), ld_value(m_consts[43])); if (!temp) { TRACE_LINE(70); break; } temp; })),t=(x?PyObject_IsTrue(x):-1),t<0) ? NULL : (t?(x):ld_value(Py_False)); if (!temp) { TRACE_LINE(70); break; } temp; })), NULL); if (!temp) { TRACE_LINE(70); break; } temp; }), ld_value(m_consts[51])); if (!temp) { TRACE_LINE(70); break; } temp; }), 0, 0, NULL, NULL); if (!temp) { TRACE_LINE(70); break; } temp; })) == -1) {
      |                                                                                                                                                                 ^~~~~~~~
      |                                                                                                                                                                 ctx_consts
.pyarmor/bcc/main.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-unsequenced’
CRITICAL unknown error, please check pyarmor.error.log
ERROR    Command '['gcc', '-O3', '-Wno-unsequenced', '-fno-asynchronous-unwind-tables', '-fno-unwind-tables', '-fPIC', '-fno-stack-protector', '-c', '-o', '.pyarmor/bcc/main.c.o', '.pyarmor/bcc/main.c']' returned non-zero exit status 1.
```

**Comment by jondy** [comment]
It's a bug, I'll fix it in next release.

**Comment by jondy** [comment]
What's `gcc` version in this machine? It seems it doesn't know this option `-Wno-unsequenced`

And could you email code around line 70 in `main.py` to pyarmor@163.com? There is a dict comprehension, it results in some errors.

**Comment by ninkin** [comment]
please find the code around main.py: L70

should I change it not to use the dict comprehension? 

```
cam_line_conf = {cam.cam_id: db_session.query(Line).filter(Line.cam_id == cam.cam_id).all() for cam in cams}
```

and also check gcc version below.


 
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 
```

**Comment by jondy** [comment]
> should I change it not to use the dict comprehension?

It doesn't matter. This will be fixed in the next release.

By the way could you test it `gcc` by this way:
```
echo "#define x 0" > a.c
gcc -Wno-unsequenced -c a.c
```

I wonder if `-Wno-unsequenced` works or not

**Comment by jondy** [comment]
Fixed in 8.0.21 except `-Wno-unsequenced` issue, but I think `gcc 9` should recognize it.

**Comment by ninkin** [comment]
> By the way could you test it `gcc` by this way:
> 
> ```
> echo "#define x 0" > a.c
> gcc -Wno-unsequenced -c a.c
> ```
> 
> I wonder if `-Wno-unsequenced` works or not

it worked without any warnings.  I guess I've used wrong gcc in my environment.
